spinta/backends/postgresql/commands/write.py

```diff
@@ -20,21 +23,50 @@ async def insert(
     transaction = context.get('transaction')
     connection = transaction.connection
     table = backend.get_table(model)
-    async for data in dstream:
-        patch = commands.before_write(context, model, backend, data=data)
-        # TODO: Refactor this to insert batches with single query.
-        qry = table.insert().values(
-            _id=patch['_id'],
-            _revision=patch['_revision'],
-            _txn=transaction.id,
-            _created=utcnow(),
-        )
-        connection.execute(qry, patch)
-
-        commands.after_write(context, model, backend, data=data)
 
+    # Need to set specific amount of max errors, to prevent memory problems
+    max_error_count = 100
+    error_list = []
+    savepoint_transaction_start = connection.begin_nested()
+    rollback_full = False
+    async for data in dstream:
+        try:
+            savepoint = connection.begin_nested()
+            patch = commands.before_write(context, model, backend, data=data)
+            # TODO: Refactor this to insert batches with single query.
+            qry = table.insert().values(
+                _id=patch['_id'],
+                _revision=patch['_revision'],
+                _txn=transaction.id,
+                _created=utcnow(),
+            )
+            connection.execute(qry, patch)
+            commands.after_write(context, model, backend, data=data)
+        except exc.IntegrityError as error:
+            rollback_full = True
+            savepoint.rollback()
+            if type(error.orig) is psycopg2.errors.ForeignKeyViolation:
+                error_message = error.orig.diag.message_detail
+                error_property_name = re.search('Key \((.*?)\)', error_message).group(1).split(".")[0]
+                error_ref_id = re.search('\)=\((.*?)\)', error_message).group(1)
```

Comment:
Use precompiled regular expressions (`re.compile`). Parsing regular expression string is pretty expensive operation.